src/gui/optionsdialog.cpp

```diff
@@ -233,23 +233,15 @@ OptionsDialog::OptionsDialog(QWidget *parent)
     m_ui->hsplitter->setCollapsible(0, false);
     m_ui->hsplitter->setCollapsible(1, false);
     // Get apply button in button box
-    const QList<QAbstractButton *> buttons = m_ui->buttonBox->buttons();
-    for (QAbstractButton *button : buttons)
-    {
-        if (m_ui->buttonBox->buttonRole(button) == QDialogButtonBox::ApplyRole)
-        {
-            m_applyButton = button;
-            break;
-        }
-    }
+    m_applyButton = m_ui->buttonBox->button(QDialogButtonBox::Apply);
```

Comment:
Just
```c++
QPushButton *applyButton = m_ui->buttonBox->button(QDialogButtonBox::Apply);
connect(applyButton, &QPushButton::clicked, this, &OptionsDialog::applySettings);
```
or even
```c++
connect(m_ui->buttonBox->button(QDialogButtonBox::Apply)
    , &QPushButton::clicked, this, &OptionsDialog::applySettings);
```

Reply:
@glassez `m_applyButton` is used in multiple method definitions throughout the `optionsdialog.cpp` file. This line is where the instance variable is assigned. Shouldn't we have the variable declared on the header file?

Reply:
Oh, sorry, I misread it. You didn't introduce this variable...
So leave it as is.